sphinx/ext/intersphinx.py

```diff
@@ -192,8 +192,7 @@ def fetch_inventory(app, uri, inv):
                     uri = path.dirname(newinv)
         with f:
             try:
-                join = localuri and path.join or posixpath.join
-                invdata = InventoryFile.load(f, uri, join)
+                invdata = InventoryFile.load(f, uri, posixpath.join)
```

Comment:
This also looks wrong - posixpath should not be used to join filesystem paths on windows, which is exactly what the old code was here to prevent